src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -584,5 +585,7 @@ private void VerifyWrite(int originalDepth, Utf8JsonWriter writer)
                 ThrowHelper.ThrowJsonException_SerializationConverterWrite(ConverterBase);
             }
         }
+
+        public ConcurrentDictionary<Type, JsonPropertyInfo> RuntimePropertyCache;
```

Comment:
Having the property `Type` be the key won't work for all cases. If there are two properties that return the same `Type` but have different attributes applied (such as `[JsonConverter]` or `[JsonIgnore`]) then they will be treated the same and thus not have the correct attibutes applies. The name needs to be included in the key.